File: luda/editor/client/src/cameraAngleEditor/imageBrowser/ImageFilenameObject.ts
export type ImageFilenameObject = {
  character: string;
  pose: string;
  emotion: string;
  extension: string;
};

export function convertImageFilenameObjectToUrl(
  filenameObject: ImageFilenameObject,
): string {
  const imageFilename = `${filenameObject.character}-${filenameObject.pose}-${filenameObject.emotion}.${filenameObject.extension}`;
  return `resources/images/${imageFilename}`;
}

File: luda-editor/luda-editor-client/src/app/types/image_filename_object.rs
use super::CharacterPoseEmotion;

#[derive(Debug, Clone)]
pub struct ImageFilenameObject {
    pub character: String,
    pub pose: String,
    pub emotion: String,
    pub url: String,
}

impl ImageFilenameObject {
    pub fn into_character_pose_emotion(&self) -> CharacterPoseEmotion {
        CharacterPoseEmotion(
            self.character.clone(),
            self.pose.clone(),
            self.emotion.clone(),
        )
    }
}
